homer0/projext-plugin-webpack
# feat(project): implement the new manual Babel polyfill

### What does this PR do?

> This is the implementation of homer0/projext#77.
> 
> The package [`@babel/polyfill`](https://yarnpkg.com/en/package/@babel/polyfill) has been deprecated in favor of using [`core-js`](https://yarnpkg.com/en/package/core-js) and [`regenerator-runtime`](https://yarnpkg.com/en/package/rengerator-runtime) directly


The project now requires `core-js`, `regenerator-runtime` and adds a custom `polyfill` file that imports them, following the recommandation of `@babel/polyfill`'s deprecation note.

This is breaking beause `WebpackConfiguration` now requires `webpackPluginInfo` in order to create the path to the "polyfill" file.

### How should it be tested manually?

In order to test this, you need to install `homer0/projext#next` and make sure you delete the copy of `projext` `npm`/`yarn` will install inside this package's `node_modules`.

Then run/build your target with `babel.polyfill` set to `true`, you should have the new polyfills in place.


Also, don't forget...

```bash
yarn test
# or
npm test
```
